### PR TITLE
Add py-scale-codec

### PR DESCRIPTION
**Pull request by MementoRC**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by MementoRC** [comment]
@conda-forge/help-python, Hello, ready for review

File: recipes/scalecodec/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - setuptools
+
+test:
+  source_files:
+    - test
+    - scalecodec/type_registry
+  requires:
+    - pytest
+  commands:
+    - pytest -v test
```
Comment (carterbox) [inline recipes/scalecodec/meta.yaml:39]:
```suggestion
test:
  source_files:
    - test
    - scalecodec/type_registry
  requires:
    - pytest
    - pip
  commands:
    - pip check
    - pytest -v test
```

File: recipes/scalecodec/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . --use-pep517 -vvv
+  script_env:
+      - GITHUB_REF=refs/tags/v{{ version }}
+  number: 0
+
+requirements:
+  host:
+    - python >=3.8.0
```
Comment (carterbox) [inline recipes/scalecodec/meta.yaml:23]: This does not match upstream.

https://github.com/polkascan/py-scale-codec/blob/4dbd98a7b66a0de94a18c315d4b678343968884b/setup.py#L172

File: recipes/scalecodec/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    - pip
+  run:
+    - base58
+    - more-itertools
+    - requests
+    - python >=3.8.0
+    - setuptools
```
Comment (carterbox) [inline recipes/scalecodec/meta.yaml:30]: https://github.com/polkascan/py-scale-codec/blob/4dbd98a7b66a0de94a18c315d4b678343968884b/setup.py#L180C1-L181C1

File: recipes/scalecodec/LICENSE [hunk 4]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (carterbox) [inline recipes/scalecodec/LICENSE:1]: You don't need to include an extra copy of the license here. LICENSE is included in the source tar.

**Review by carterbox — CHANGES_REQUESTED** [review]
Why are you trying to publish duplicate packages? It seems that there is not a non-python version of scalecodec, so you should just have one recipe for `scalecodec`.

**Comment by MementoRC** [comment]
@conda-forge/help-python ready for review

File: recipes/pypromice/meta.yaml [hunk 5]
```diff
@@ -1,69 +0,0 @@
-{% set name = "pypromice" %}
-{% set version = "1.3.1" %}
```
Comment (carterbox) [inline recipes/pypromice/meta.yaml:2]: Please don't delete other people's packages from the recipes directory. It causes merge conflicts in our main branch. We have a bot which deletes packages as they are converted.

File: recipes/scalecodec/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+{% set repo_name = "py-scale-codec" %}
+{% set version = "1.2.7" %}
+{% set sha256 = "d5723891dba332200017e881ab839c01a7b035863df1585a524c209cfcea5aed" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/polkascan/{{ repo_name }}/archive/refs/tags/v{{ version }}.tar.gz
```
Comment (carterbox) [inline recipes/scalecodec/meta.yaml:11]:
```suggestion
{% set name = "scalecodec" %}
{% set version = "1.2.7" %}
{% set sha256 = "d5723891dba332200017e881ab839c01a7b035863df1585a524c209cfcea5aed" %}

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://github.com/polkascan/py-scale-codec/archive/refs/tags/v{{ version }}.tar.gz
```

File: recipes/scalecodec/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - base58 >=2.0.1
+    - more-itertools
+    - requests >=2.24.0
+    - python >=3.6
+    - setuptools
```
Comment (carterbox) [inline recipes/scalecodec/meta.yaml:30]: Upstream doesn't list setuptools in the runtime requirements. Maybe you just wanted it in the host requirements?

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/scalecodec/meta.yaml (hunk 7 above)
Comment (MementoRC) [reply recipes/scalecodec/meta.yaml:30]: Will do

File: recipes/pypromice/meta.yaml (hunk 5 above)
Comment (MementoRC) [reply recipes/pypromice/meta.yaml:2]: Will close and resubmit the PR

Comment (carterbox) [reply recipes/pypromice/meta.yaml:2]: You can use `git rebase -i`

Comment (MementoRC) [reply recipes/pypromice/meta.yaml]: @carterbox Is this still needed?

**Comment by MementoRC** [comment]
@carterbox I am not sure the `pyromice` issue was solved. `substrate-interface` require this package and recently both upstream were updated, it seems more practical to work on both at the same time at my end, so I combined them and submitted a new PR. Let me know if this is not convenient

New PR: https://github.com/conda-forge/staged-recipes/pull/25470

**Comment by carterbox** [comment]
> so I combined them and submitted a new PR. Let me know if this is not convenient

This is an annoying behavior. You could have just pushed changes to the MementoRC:scalecodec branch. The scalecodec part of the PR now needs to be re-reviewed.

**Comment by MementoRC** [comment]
Ok, then I re-open and close the new one. I just felt it ws simpler for both @xhochy and yourself to finalize a bundled recipe. I sure did work locally on both, but I only have a linux machine - far from me to generate such annoyance - though, I guess I can do no right at this point

**Comment by MementoRC** [comment]
@arjanz Please comment here at your convenience

**Review by carterbox — APPROVED** [review]
Just waiting on comment from all maintainers.

**Comment by arjanz** [comment]
> @arjanz Please comment here at your convenience

Hi, I'm the maintainer of [py-scale-codec](https://github.com/polkascan/py-scale-codec), so makes sense to be in the loop

**Comment by MementoRC** [comment]
Close/reopen to trigger linter

**Pull request merged**